pkgs/build-support/go/module.nix

```diff
@@ -338,8 +344,18 @@ lib.extendMkDerivation {
                   export NIX_BUILD_CORES=1
               fi
               for pkg in $(getGoDirs ""); do
-                echo "Building subPackage $pkg"
-                buildGoDir install "$pkg"
+                ${
+                  if buildTestBinaries then
+                    ''
+                      echo "Building test binary for $pkg"
+                      buildGoDir "test -c -o $GOPATH/bin/" "$pkg"
```

Comment:
It would have been nice to add this to `GOFLAGS` in the derivation, but as there is no variable substitution for `GOFLAGS`, `$GOPATH` would be used literally.